cosmos_tls_scanner.py

```diff
@@ -0,0 +1,216 @@
+
+
+def _send_request(
+    uri, ssl_version, verb, resource_type, resource_link, headers=None, body=None
+):
+    """ Send REST call with given ssl_version """
+    full_url = urllib.parse.urljoin(uri, resource_link + "/")
+    full_url = urllib.parse.urljoin(full_url, resource_type)
+    session = requests.Session()
+    session.mount("https://", SSLAdapter(ssl_version))
+    response = session.request(verb, full_url, headers=headers, data=body)
+    return response
+
+
+def list_databases(uri, key, ssl_version):
+    """ Basic call to list all databases in an account """
+    verb = "get"
+    resource_type = "dbs"
+    resource_link = ""
+    date = datetime.datetime.utcnow()
+    date_str = date.strftime("%a, %d %b %Y %H:%M:%S GMT").lower()
+
+    headers = {
+        "authorization": _get_auth_header(
+            key, verb, resource_type, resource_link, date_str
+        ),
+        "x-ms-version": "2017-02-22",
+        "x-ms-date": date_str,
+        "x-ms-max-item-count": "1",
+        "Cache-Control": "no-cache",
+    }
+    return _send_request(uri, ssl_version, verb, resource_type, resource_link, headers)
+
+
+def basic_query(uri, key, ssl_version, database_name, container_name):
+    """ Basic call to query one element from given database, container """
+    verb = "post"
+    resource_type = "docs"
+    resource_link = "dbs/{}/colls/{}".format(database_name, container_name)
+    date = datetime.datetime.utcnow()
+    date_str = date.strftime("%a, %d %b %Y %H:%M:%S GMT").lower()
+
+    headers = {
+        "content-type": "application/query+json",
+        "authorization": _get_auth_header(
+            key, verb, resource_type, resource_link, date_str
+        ),
+        "x-ms-version": "2017-02-22",
+        "x-ms-date": date_str,
+        "x-ms-documentdb-isquery": "True",
+        "Cache-Control": "no-cache",
+    }
+
+    body = '{"query":"SELECT 1"}'
+    return _send_request(
+        uri, ssl_version, verb, resource_type, resource_link, headers, body
+    )
+
+
+def _get_parser():
+    """
+      Gets argparse parser object
+      Args:
+        --endpoint: Database account endpoint. Example https://myaccount.documents.azure.com:443/
+        --authorization-key: Master or Read-only key for account of the form khYANAIiAl12n...==
+        --database-name: Name of the CosmosDB database in the account.
+        --collection-name: Name of the collection in the database.
+    """
+    parser = argparse.ArgumentParser(description="CosmosDB TLS Scanner")
+    parser.add_argument(
+        "--endpoint",
+        "-e",
+        required=True,
+        help="CosmosDB database account endpoint. Example https://myaccount.documents.azure.com:443/",
+    )
+    parser.add_argument(
+        "--authorization-key",
+        "-k",
+        required=True,
+        help="Master or Read-only key for account of the form khYANAIiAl12n...==",
+    )
+    parser.add_argument(
+        "--database-name",
```

Comment:
we need to trim any trailing or ending whitespace on every parameter.

Reply:
Already stripping arguments of whitespaces in the beginning of main function